src/GafferSceneUI/Inspector.cpp

```diff
@@ -231,19 +278,57 @@ void Inspector::inspectHistoryWalk( const GafferScene::SceneAlgo::History *histo
 
 					if( !result->m_editScope || node->ancestor<EditScope>() == result->m_editScope )
 					{
-						const GraphComponent *readOnlyReason = MetadataAlgo::readOnlyReason( result->m_source.get() );
-						if( !result->m_editScope && readOnlyReason )
+						std::string nonEditableReason;
+						if( auto n = runTimeCast<NameValuePlug>( result->m_source.get() ) )
 						{
-							result->m_editFunction = fmt::format(
-								"{} is locked.",
-								readOnlyReason->relativeName( readOnlyReason->ancestor<ScriptNode>() )
+							nonEditableReason = lockedReason( n, result->m_editScope.get() );
+							if( nonEditableReason.empty() )
+							{
+								nonEditableReason = ::nonEditableReason( n->valuePlug(), result->m_editScope.get() );
+							}
+							if( nonEditableReason.empty() )
+							{
+								nonEditableReason = ::nonEditableReason( n->enabledPlug(), result->m_editScope.get() );
+							}
+						}
+						else if( auto t = runTimeCast<TweakPlug>( result->m_source.get() ) )
```

Comment:
At this point, Inspectors have no reason to know anything about TweakPlugs or NameValuePlugs. I think it would be better if they simply considered the editability of `m_source` and all its descendants, without special cases, and if all that was done for it by `nonEditableReason()`. I presume that would also be needed for other compound types, like Color3fPlug.

The only difference I can see is that here we're not considering the `TweakPlug.mode`, but I don't think it would do much harm to consider that?

Reply:
> but I don't think it would do much harm to consider that

I don't expect it to be an issue either. Checking recursively is done in https://github.com/GafferHQ/gaffer/pull/5395/commits/73056bcd8ed46fabb4fd04b5989bb683f8adbc2d